roles/rsyslog/tasks/main.yml

```diff
@@ -90,9 +92,28 @@
             state: absent
             path: "{{ __rsyslog_config_dir }}/*"
           when: __rsyslog_purge_original_conf | bool | d(false)
+
+        - block:
+            - name: Get backup files
+              shell: ls "{{ __rsyslog_backup_dir }}"
+              register: __backups
+
+            - name: Get backup file stat
+              stat:
+                path: "{{ __rsyslog_backup_dir }}/{{ __backups.stdout_lines[0] }}"
+              register: __backup_stat
+
+            - name: Restore original conf
+              shell: |-
+                # set -euo pipefail
+                tar -xvzPf "{{ __rsyslog_backup_dir }}/{{ __backups.stdout_lines[0] }}"
```

Comment:
use `command` instead of `shell`?

Reply:
is `__backups.stdout_lines[0]` the earliest file or the latest file?

Reply:
> is `__backups.stdout_lines[0]` the earliest file or the latest file?

The earliest. I was hoping to restore from the one prior to running the first rhel-system-roles.logging. That is, the one from the original rsyslog rpm, if available.

Rethinking of it, do we want to have two variables `logging_restore_confs_from_prev` and `logging_restore_confs_from_first` instead of one?

Reply:
> > is `__backups.stdout_lines[0]` the earliest file or the latest file?
> 
> The earliest. I was hoping to restore from the one prior to running the first rhel-system-roles.logging. That is, the one from the original rsyslog rpm, if available.

So we really only need 1 backup - something like
```
if not backup exists in backup_dir; then
  create backup in backup_dir
fi
```
The problem with creating an indefinite number of backups is that there is no way to clean them up.

> 
> Rethinking of it, do we want to have two variables `logging_restore_confs_from_prev` and `logging_restore_confs_from_first` instead of one?

No.  I think just "restore_confs" meaning "restore to original state" is sufficient.